pkgs/by-name/bo/bochs/options.nix

```diff
@@ -0,0 +1,17 @@
+{
+  lib,
+  config,
+  stdenv,
+  ...
+}:
+
+{
+  options = {
+    interfaces = lib.mkOption {
+      default = [ "SDL2" "term" "wx" "x11" ];
```

Comment:
Lists compose quite poorly. A better design could be to make this an attrset or individual options such that you can set them like this:

```Nix
{
  sdl2 = true;
  term = true;
  wx = true;
  x11 = true;
  somethinElse = false;
}
```

Reply:
> a bunch of individual options

Like the previous commit?
https://github.com/NixOS/nixpkgs/pull/342188/commits/0e3afc49b1f4a4bb24235fa3f3c14d7b905b3005

> a freeform attrset

Like this?

```nix
options = {
  interfaces = lib.mkOption {
    default = {
      sdl =  true;
      term =  true;
      # . . .
    };
  };
};
```

Reply:
I like to use Bochs expression as a "test+documentation" example.
I wanna try that settings-like approach just for the docs.